hooks/gate-scripts/lib/commit_scope.py

```
@@ -0,0 +1,359 @@
+    `enabledPlugins`/`extraKnownMarketplaces`, which reach the same capability by
+    activating a plugin whose manifest registers the hook. Any `env` at all — an
+    allowlist would have to track every variable git and sh read.
+    """
+    for name in ("settings.json", "settings.local.json"):
+        path = os.path.join(repo, state_dir, name)
+        if not os.path.exists(path):
+            continue
+        try:
+            with open(path, "rb") as fh:
+                data = json.loads(fh.read().decode("utf-8", "surrogateescape"))
+        except Exception:
```

Ruff (0.16.1)

[warning] Do not catch blind exception: `Exception`

(BLE001)